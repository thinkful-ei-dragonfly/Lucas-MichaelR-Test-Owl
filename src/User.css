/* Participant List */
.User {
  display: flex;
  align-items: flex-start;
  flex-direction: row;
}
.image-wrapper {
  width: 50px;
  margin-right: 10px;
}
.image-wrapper img {
  max-width: 100%;
}
.user-info h4{
  margin: 0.75em 0 0.5em 0;
}
.user-status {
  margin-top: 0;
}
.user-status::before {
  width: 10px;
  height: 10px;
  background: lightgreen;
  border-radius: 50%;
  content: " ";
  display: inline-block;
  margin-right: 7.5px;
}

/* Stage User */
.stage-user {
  width: 50%;
  text-align: center;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.stage-user img {
  max-width: 75%;
  margin: 0 auto;
  width: 100%;
}
.stage-user-info {
  background: rgba(30,40,53,.65);
  color: white;
  font-size: 18px;
  font-weight: medium;
  position: absolute;
  top: 0;
  left: 33%;
  right: 33%;
  border-bottom-right-radius: 10px;
  border-bottom-left-radius: 10px;
}
.stage-user-name {
  padding-bottom: 5px;
  border-bottom: 3px solid #f37a7b;
  border-radius: 2px;
}
.volume {
  margin-right: 10px;
}
