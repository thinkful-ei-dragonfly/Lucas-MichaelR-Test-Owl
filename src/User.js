import React from 'react';
import { directive } from '@babel/types';

function User(props) {
  let activeStatus = 'in session';
  if (props.onStage) {
    activeStatus = 'on stage';
  };
  if (props.display === 'inList') {
    return <div className="User">
      <div className="image-wrapper">
        <img src={props.avatar} alt="user_avatar"/>
      </div>>
      <div className="user-info">
        <h4>{props.name}</h4>
        <p className="user-status">{activeStatus}</p>
      </div>
    </div>
  } else if (props.display === 'onStage') {
    return <div>Stage user</div>
  }
}

export default User;