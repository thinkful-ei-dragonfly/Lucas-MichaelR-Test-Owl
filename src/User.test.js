import React from 'react';
import renderer from 'react-test-renderer';

import User from './User';

const user = {
  id: 1,
  name: 'Koren Templeton',
  avatar:
      'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
  inSession: true,
  onStage: true
}

describe('<User  />', () => {
  it('renders without crashing [snapshot test]', () => {
    // Render this component as JSON
    const tree = renderer.create(<User key={user.id} display={'inList'}avatar={user.avatar} name={user.name} onStage={user.onStage}/>).toJSON();
    // Checks wether this matches the previous snapshot
    // The previous snapshot will be stored in __snapshots__/User .test.js
    expect(tree).toMatchSnapshot();
  })
})