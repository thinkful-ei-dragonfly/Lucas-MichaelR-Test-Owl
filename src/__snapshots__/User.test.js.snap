// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User  /> renders without crashing [snapshot test] 1`] = `
<div
  className="User"
>
  <div
    className="image-wrapper"
  >
    <img
      alt="user_avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="user-info"
  >
    <h4>
      Koren Templeton
    </h4>
    <p
      className="user-status"
    >
      on stage
    </p>
  </div>
</div>
`;
